application/src/main/java/com/musicforall/services/recommendation/RecommendationService.java

```diff
@@ -21,9 +25,13 @@
  * Created by kgavrylchenko on 16.08.16.
  */
 @Component
+@Transactional
 public class RecommendationService {
```

Comment:
RecommendationService was not designed to be transactional, its just spring bean that works with services using composition. so, remove Transactional annotation